pkg/tests/tasks/security/authorization/ext_auth_test.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2021 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package authorization
+
+import (
+	"testing"
+
+	"github.com/maistra/maistra-test-tool/pkg/app"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
+	"github.com/maistra/maistra-test-tool/pkg/util/hack"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/pod"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/test"
+)
+
+func TestExtAuth(t *testing.T) {
+	test.NewTest(t).Id("T37").Groups(test.Full, test.InterOp).Run(func(t test.TestHelper) {
+		hack.DisableLogrusForThisTest(t)
+
+		ns := "foo"
+		t.Cleanup(func() {
+			oc.RecreateNamespace(t, ns)
+			oc.Patch(t, meshNamespace,
+				"smcp", smcpName,
+				"json",
+				`[{"op": "remove", "path": "/spec/techPreview"}]`)
```

Comment:
Why do you patch SMCP instead of applying the expected YAML?

Reply:
Here, I'm removing the patch. 

I have patched this https://github.com/maistra/maistra-test-tool/blob/f5d70f54e078e8f914153b6a692842d700f82603/pkg/tests/tasks/security/authorization/ext_auth_test.go#L68, At the cleaning state. I'm removing what we patched. 


i was tested, external authorization is removed successfully from the SMCP.

Reply:
The expected state is:
```
spec:
  externalProviders:
  - name: sample-ext-authz-http
    envoyExtAuthzHttp:
      service: ...
      port: ...
      includeRequestHeadersInCheck: ...
```

So we can handle this problem by always using the same name and namespace.